buildpacks/conda-nodefaults/bin/build

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+echo -e "🔧 \033[1mConda Channel Cleanup Buildpack\033[0m"
+
+ENV_FILE="environment.yml"
+
+# Exit silently if environment.yml is not present
+[[ -f "$ENV_FILE" ]] || exit 0
+
+# Save original version for diff
+TMPDIR=$(mktemp -d)
+cp environment.yml "$TMPDIR/original.yml"
+
+# Download latest yq binary
+OS=$(uname -s | tr '[:upper:]' '[:lower:]')
+ARCH=$(uname -m)
+
+case "$ARCH" in
+  x86_64) ARCH=amd64 ;;
+  arm64|aarch64) ARCH=arm64 ;;
+  *) echo "Unsupported architecture: $ARCH"; exit 1 ;;
+esac
+
+YQ_FILENAME="yq_${OS}_${ARCH}"
+YQ_URL="https://github.com/mikefarah/yq/releases/latest/download/${YQ_FILENAME}"
+YQ_BIN="$TMPDIR/yq"
+
+curl -fsSL "$YQ_URL" -o "$YQ_BIN"
+chmod +x "$YQ_BIN"
+
+# Ensure 'channels' exists (set to empty if missing)
+"$YQ_BIN" eval -i '.channels = (.channels // [])' $ENV_FILE
+
+# Fail if "defaults" is in the channels list
+if "$YQ_BIN" '.channels | any_c(. == "defaults")' $ENV_FILE | grep -q true; then
```

Comment:
suggestion: This grep is a bit risky.

You can add `--exit-status` on yq as a global flag which will set the exit code to non-zero if the result is false or not found.

>  -e, --exit-status                   set exit status if there are no matches or null or false is returned